http/sys_raft.go

```diff
@@ -27,7 +29,31 @@ func handleSysRaftJoinPost(core *vault.Core, w http.ResponseWriter, r *http.Requ
 		return
 	}
 
-	joined, err := core.JoinRaftCluster(context.TODO(), req.LeaderAddr, nil, req.Retry)
+	if req.LeaderAddr == "" {
+		respondError(w, http.StatusBadRequest, errors.New("leader api address is required"))
+		return
+	}
+
+	config := api.DefaultConfig()
+	config.Address = req.LeaderAddr
+	if req.CACert != "" && req.ClientCert != "" && req.ClientKey != "" {
+		tlsConfig := &api.TLSConfig{
+			CACert:     req.CACert,
+			ClientCert: req.ClientCert,
+			ClientKey:  req.ClientKey,
+		}
+		if err := config.ConfigureTLS(tlsConfig); err != nil {
```

Comment:
I believe this only works with paths to Cert/Key files. instead we want to provide the full config (path wont exist on the server)

Reply:
Fixed.